task/control-plane-agent/src/update/host_flash.rs

```diff
@@ -29,6 +29,42 @@ impl HostFlashUpdate {
             current: None,
         }
     }
+
+    pub(crate) fn active_slot(&self) -> Result<u16, SpError> {
+        match self
+            .task
+            .get_dev()
+            .map_err(|err| SpError::ComponentOperationFailed(err as u32))?
+        {
+            HfDevSelect::Flash0 => Ok(0),
+            HfDevSelect::Flash1 => Ok(1),
+        }
+    }
+
+    pub(crate) fn set_active_slot(&self, slot: u16) -> Result<(), SpError> {
+        let slot = match slot {
+            0 => HfDevSelect::Flash0,
+            1 => HfDevSelect::Flash1,
+            _ => return Err(SpError::InvalidSlotForComponent),
+        };
+
+        // Do we have control of the host flash?
+        match self
+            .task
+            .get_mux()
+            .map_err(|err| SpError::UpdateFailed(err as u32))?
+        {
+            HfMuxState::SP => (),
+            HfMuxState::HostCPU => return Err(SpError::UpdateSlotBusy),
+        }
+
+        // Swap to the chosen slot.
+        self.task
+            .set_dev(slot)
+            .map_err(|err| SpError::UpdateFailed(err as u32))?;
```

Comment:
It seems like there's a TOCTOU race here.

Luckily, `self.task.set_dev(slot)` _also_ checks that the mux is valid; we can probably eliminate the first check and just its its result directly

```suggestion
        // Swap to the chosen slot, returning an appropriate error if we don't
        // have control over the host flash.
        self.task
            .set_dev(slot)
            .map_err(|err| match err{
                HfError::NotMuxedToSP => SpError::UpdateSlotBusy,
                _ => SpError::UpdateFailed(err as u32)
            })?;
```

(untested)

Reply:
Nice catch, fixed.